front/lib/api/workspace.ts

```diff
@@ -117,3 +153,300 @@ export async function getPendingInvitations(
+              await RetrievalDocumentChunk.destroy({
+                where: {
+                  retrievalDocumentId: retrievalDocument.id,
+                },
+                transaction: t,
+              });
+              await retrievalDocument.destroy({ transaction: t });
+            }
+            await AgentRetrievalAction.destroy({
+              where: { id: agentMessage.agentRetrievalActionId },
+              transaction: t,
+            });
+          }
+          await agentMessage.destroy({ transaction: t });
+        }
+      }
+      if (msg.contentFragmentId) {
+        await ContentFragment.destroy({
+          where: { id: msg.contentFragmentId },
+          transaction: t,
+        });
+      }
+      await MessageReaction.destroy({
+        where: { messageId: msg.id },
+        transaction: t,
+      });
+      await Mention.destroy({
+        where: { messageId: msg.id },
+        transaction: t,
+      });
+      await msg.destroy({ transaction: t });
+    }
+    await ConversationParticipant.destroy({
+      where: { conversationId: convo.id },
+      transaction: t,
+    });
+    await convo.destroy({ transaction: t });
+  }
+};
+
+const _deleteAgents = async (workspace: WorkspaceType, t: Transaction) => {
+  const agents = await AgentConfiguration.findAll({
+    where: {
+      workspaceId: workspace.id,
+    },
+  });
+  for (const agent of agents) {
+    if (agent.generationConfigurationId) {
+      await AgentGenerationConfiguration.destroy({
+        where: {
+          id: agent.generationConfigurationId,
+        },
+        transaction: t,
+      });
+    } else if (agent.retrievalConfigurationId) {
+      await AgentDataSourceConfiguration.destroy({
+        where: {
+          retrievalConfigurationId: agent.retrievalConfigurationId,
+        },
+        transaction: t,
+      });
+      await AgentRetrievalConfiguration.destroy({
+        where: {
+          id: agent.retrievalConfigurationId,
+        },
+        transaction: t,
+      });
+    } else if (agent.dustAppRunConfigurationId) {
+      await AgentDustAppRunAction.destroy({
+        where: {
+          dustAppRunConfigurationId: agent.dustAppRunConfigurationId,
+        },
+        transaction: t,
+      });
+      await AgentDustAppRunConfiguration.destroy({
+        where: {
+          id: agent.dustAppRunConfigurationId,
+        },
+        transaction: t,
+      });
+    }
+    await MemberAgentVisibility.destroy({
+      where: {
+        agentConfigurationId: agent.id,
+      },
+      transaction: t,
+    });
+    await agent.destroy({ transaction: t });
+  }
+  await GlobalAgentSettings.destroy({
+    where: {
+      workspaceId: workspace.id,
+    },
+    transaction: t,
+  });
+};
+
+const _deleteApps = async (workspace: WorkspaceType, t: Transaction) => {
+  const apps = await App.findAll({
+    where: { workspaceId: workspace.id },
+    transaction: t,
+  });
+  for (const app of apps) {
+    await Run.destroy({
```

Comment:
I added a commit to throw an error if we're trying to delete a workspace with Apps. So this could be merged until I figured out the implementation of a `deleteProject` in Core.